GVFS/GVFS.Platform.Mac/MacFileSystem.cs

```diff
@@ -36,6 +36,41 @@ public void ChangeMode(string path, int mode)
         public bool TryGetNormalizedPath(string path, out string normalizedPath, out string errorMessage)
         {
             return MacFileSystem.TryGetNormalizedPathImplementation(path, out normalizedPath, out errorMessage);
+        }
+
+        public bool TryReadFirstByteOfFile(string fileName)
+        {
+            return NativeFileReader.TryReadFirstByteOfFile(fileName);
+        }
+
+        private class NativeFileReader
+        {
+            public const int ReadOnly = 0x0000;
+
+            public static bool TryReadFirstByteOfFile(string fileName)
+            {
+                int fileDescriptor = Open(fileName, ReadOnly);
+                return TryReadOneByte(fileDescriptor);
+            }
+
+            private static bool TryReadOneByte(int fileDescriptor)
+            {
+                byte[] buffer = new byte[1];

```

Comment:
We'll actually need to fix this pointer before passing it over to the native code. Otherwise the risk is that the GC can move this memory while the native code is running, and it will write to a now invalid pointer